gnpy/core/equipment.py

```diff
@@ -0,0 +1,119 @@
+        g1a_min = gain_min - (gain_max-gain_min) - delta_p
+        if delta_p < 1 or delta_p > 6:
+            #delta_p should be > 1dB and < 6dB => consider user warning if not
+            print('1st coil vs 2nd coil calculated DeltaP {} is not valid: revise inputs'
+                        .format(delta_p))
+    #check the calculated values for nf1 & nf2:
+    nf_min_calc = lin2db(db2lin(nf1) + db2lin(nf2)/db2lin(g1a_max))
+    nf_max_calc = lin2db(db2lin(nf1) + db2lin(nf2)/db2lin(g1a_min))
+    if (abs(nf_min_calc-nf_min) > 0.01) or (abs(nf_max_calc-nf_max) > 0.01):
+        print('nf model calculation failed with nf_min {} and nf_max {} calculated'
+                .format(nf_min_calc, nf_max_calc))
+
+    return (nf1, nf2, delta_p)
+
+
+def read_eqpt_library(filename):
+    """build global dictionnary eqpt_library that stores all eqpt characteristics:
+    edfa type type_variety, fiber type_variety
+    from the eqpt_config.json (filename parameter)
+    also read advanced_config_from_json file parameters for edfa if they are available:
+    typically nf_ripple, dfg gain ripple, dgt and nf polynomial nf_fit_coeff
+    if advanced_config_from_json file parameter is not present: use nf_model:
+    requires nf_min and nf_max values boundaries of the edfa gain range
+    """
+    global eqpt_library
+    eqpt_library = load_json(filename)
+    for i, el in enumerate(eqpt_library['Edfa']): 
+        dict_nf_model = {}
+        if 'advanced_config_from_json' in el:
+            #use advanced amplifier model with full ripple characterization
+            config_json_filename = el.pop(ADVANCED_CONFIG_JSON_FILENAME)
+            dict_nf_model['nf_model'] = {'enabled': False}
+        else:
+            #use a default ripple model (only default dgt is defined)
+            config_json_filename = DEFAULT_CONFIG_JSON_FILENAME
+            (nf1, nf2, delta_p) = nf_model(el)
+            #remove nf_min and nf_max field and replace by nf1, nf2 & delta_p
+            nf_min = el.pop('nf_min','')
+            nf_max = el.pop('nf_max','')
+            dict_nf_model['nf_model'] = dict(zip(["enabled","nf1","nf2","delta_p"],[True,nf1,nf2,delta_p]))
+
+        json_data = load_json(config_json_filename)
+        eqpt_library['Edfa'][i] = {**el, **json_data, **dict_nf_model}
+
+
+def get_eqpt_params(eqpt_name):
+    """returns the params attributs of an Edfa or Fiber 
+    by finding it in the eqpt_library
+    input parameter eqpt_name = type_variety of the eqpt
+    """
+    eqpt_config={}
+    try:
+        #go through the eqpt library to find the demanded type_variety
+        eqpt = next(eqpt for eqpt_type in eqpt_library
+                    for eqpt in eqpt_library[eqpt_type] 
+                    if eqpt.get('type_variety','') == eqpt_name)
+        eqpt_config = dict(eqpt)
+        del eqpt_config['type_variety']
+    except StopIteration as e:
+        print(f'cannot find eqpt {eqpt_name} in eqpt library')
+        #TODO log it
+    return eqpt_config
```

Comment:
This whole lookup should be unnecessary.

The equipment library should just be a dict from which you can query the equipment you want by type/subtype.

Reply:
Strongly disagree.
If the topology parser calls for an eqpt that doesn't exist it should be logged out
=> it will be a very common mistake (spelling, libraires differences...)